active/tab/tabcontrol.md

```diff
@@ -0,0 +1,580 @@
+from their name.  These descriptions are what show up
+in IntelliSense. -->
+
+### TabView properties, events, methods
+
+| Property | Description |
+|:-------- |:----------- |
+| AddTabButtonCommand | Gets or sets the command to invoke when the Add button is tapped. |
+| AddTabButtonCommandParameter | Gets or sets the parameter to pass to the command for the Add button. |
+| CanDragTabs | Gets or sets a value that indicates whether tabs in the collection can be dragged. Default is true. |
+| CanReorderTabs | Gets or sets a value that indicates whether tabs in the collection can be reordered through user interaction. Default is true. |
+| IsAddTabButtonVisible | Determines if the plus button appears to the right of the tab strip |
+| TabItemsSource | Gets or sets an object source used to generate the tabs. |
+| TabItems | Gets the collection used to generate the tabs. |
+| ItemTemplate | Gets or sets the DataTemplate used to display each item. |
+| ItemTemplateSelector | Gets or sets a reference to a custom DataTemplateSelector logic class. The DataTemplateSelector referenced by this property returns a template to apply to items. |
+| SelectedIndex | Gets or sets the index of the selected item. |
+| SelectedItem | Gets or sets the selected item. |
+| TabStripHeader | Content to the left of the tab strip. |
+| TabStripFooter | Content to the right of the tab strip. |
+| TabViewItemWidthMode | Specifies how the tabs should be sized. Values are {Actual, Equal}. Default is Actual. |
+
+| Event | Description |
+|---|---|
+| AddTabButtonClick | Fires when the + button is clicked. |
+| SelectionChanged | Occurs when the currently selected tab changes. |
+| TabClosing | Fires when a tab is about to be closed. Can be cancelled to prevent closure. |
+| TabDraggedOutside | Fires when a Tab is dragged outside of the Tab bar. |
+
+| Method | Description |
+|---|---|
+| DependencyObject ContainerFromItem(object item) | Returns the generated container for the specified item. |
+| DependencyObject ContainerFromIndex(Int32 index) | Returns the generated container for the specified index. |
+
+### TabViewItem properties and events
+
+| Property | Description |
+|:-------- |:----------- |
+| Content | The main content that appears in the tab. |
+| Header | The content that appears inside the tab itself.  |
+| HeaderTemplate | Template for the header object. |
+| Icon | Icon for the tab. |
+| IsCloseable | Determines if the tab shows a close button. Default is "True". |
+
+| Event | Description |
+|---|---|
+| TabClosing | Fires when a tab's close button is clicked. |
+
+# API Details
+<!-- The exact API, in MIDL3 format (https://docs.microsoft.com/en-us/uwp/midl-3/) -->
+
+```
+namespace MU_XC_NAMESPACE
+{
+
+[WUXC_VERSION_PREVIEW]
+[webhosthidden]
+enum TabViewItemWidthMode
+{
+    Actual = 0,
+    Equal = 1,
+};
+
+[WUXC_VERSION_PREVIEW]
+[webhosthidden]
+runtimeclass TabViewTabClosingEventArgs
+{
+    Object Item { get; };
+
+    Boolean Cancel { get; set; };
```

Comment:
If I don't cancel, does the TabView automatically remove the tab and destroy it?

What if I need to prompt "Save changes?" Do we need a deferral?

Maybe we don't need a Cancel at all. This is just a `TabViewTabCloseRequestedEventArgs`, and it's up to the app to deal with the close request. Maybe it removes the tab immediately. Maybe it prompts before removing the tab. Does the system need to know when the close request is complete?

Reply:
Discussion: note that the dropped outside event requires the app to remove the tab item, but this event does not.

Recommend: Remove the Cancel property. Don't actually remove the tab item from the control; leave it for the app.

Reply:
Should we rename TabClosing to something else (like CloseTabRequested)?

Reply:
Yes, definitely.